src/SonarScanner.MSBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -65,17 +64,24 @@ public async Task<ISonarWebServer> CreateSonarWebServer(ProcessedArgs args, IDow
                 return null;
             }
 
-            downloader ??= new WebClientDownloaderBuilder(args.ServerInfo.ServerUrl, args.HttpTimeout, logger)
-                            .AddAuthorization(userName, password)
-                            .AddCertificate(clientCertPath, clientCertPassword)
-                            .Build();
+            downloader ??= Downloader(args.ServerInfo.ServerUrl);
+            apiDownloader ??= Downloader(args.ServerInfo.ApiBaseUrl);
 
-            var serverVersion = await QueryServerVersion(downloader);
-            if (serverVersion == null)
+            var serverVersion = await QueryServerVersion(apiDownloader, downloader);
```

Comment:
This method is getting too big, and I think it's time to split it. You can extract one or more validation methods.

Reply:
Let me do it on the next PR, since I will be touching this part again